toolchain/lexer/tokenized_buffer_test_helpers.h

```diff
@@ -58,7 +58,7 @@ struct ExpectedToken {
   int indent_column = -1;
   bool recovery = false;
   llvm::StringRef text = "";
-  llvm::Optional<llvm::StringRef> string_contents = llvm::None;
+  std::optional<llvm::StringRef> string_contents = llvm::None;
```

Comment:
If we're switching, uses of llvm::None should also be replaced with std::nullopt

Reply:
The amazing thing is that this compiled.

The reason is because `llvm::None` is defined in terms of `std::nullopt` despite being used with `llvm::Optional`.

That said, I'm second guessing my whole approach here, much as you indicate I should from #1118 -- Maybe I should *just* update `hasValue` to `has_value`?

I'm worried there are still cases where LLVM provides useful optimized storage for `llvm::Optional` that `std::optional` won't, but curious what other LLVM knowledgeable folks think.

Reply:
From https://discourse.llvm.org/t/deprecating-llvm-optional-x-hasvalue-getvalue-getvalueor/63716/11 it looks like the overall intent is to migrate to std::optional, so it seems the intent on the LLVM side might be to make this work even if llvm::Optional is still required in some cases (MapEntryOptionalStorage is noted).

Reply:
Made #2432 to try to break out the has_value part -- note, not _instead_ of this change, but only to get a quick fix on that one piece.

Reply:
Eh, let's just land it. This PR is updated to fully remove it and I agree with your rationale if upstream is moving awy.